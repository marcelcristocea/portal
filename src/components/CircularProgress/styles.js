export default theme => ({
  circularProgressWrapper: {
    padding: 25
  },
  circularProgress: {
    color: '#4479bd',
    display: 'block',
    margin: '0 auto'
  }
});
