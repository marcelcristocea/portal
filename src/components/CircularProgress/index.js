import React from 'react';
import PropTypes from 'prop-types';
import {CircularProgress as MuiCircularProgress} from '@material-ui/core';
import {withStyles} from '@material-ui/core/styles';

import styles from './styles';

function CircularProgress({classes, size}) {
  return (
    <div className={classes.circularProgressWrapper}>
      <MuiCircularProgress size={size} className={classes.circularProgress} />
    </div>
  );
}

CircularProgress.propTypes = {
  size: PropTypes.number
};

CircularProgress.defaultProps = {
  size: 50
};

export default withStyles(styles)(CircularProgress);
