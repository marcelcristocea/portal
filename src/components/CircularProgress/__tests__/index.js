import * as React from 'react';
import renderer from 'react-test-renderer';

import CircularProgress from '../';

describe('CircularProgress', () => {
  it('renders correctly by default', () => {
    const tree = renderer
      .create(
        <CircularProgress />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
