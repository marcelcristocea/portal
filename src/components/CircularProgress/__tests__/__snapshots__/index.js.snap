// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgress renders correctly by default 1`] = `
<div
  className="CircularProgress-circularProgressWrapper-1"
>
  <div
    className="MuiCircularProgress-root-3 MuiCircularProgress-colorPrimary-6 MuiCircularProgress-indeterminate-5 CircularProgress-circularProgress-2"
    role="progressbar"
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg-8"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle-9 MuiCircularProgress-circleIndeterminate-11"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
